lib/chef/knife/cookbook_metadata.rb

```diff
@@ -93,7 +93,10 @@ def generate_metadata_from_file(cookbook, file)
       def validate_metadata_json(path, cookbook)
         json_file = File.join(path, cookbook, "metadata.json")
         if File.exist?(json_file)
-          Chef::Cookbook::Metadata.validate_json(IO.read(json_file))
+          ui.info("Generating metadata for #{cookbook} from #{json_file}")
+          md = Chef::Cookbook::Metadata.new
+          md.name(cookbook)
+          md.validate_json(IO.read(json_file))
```

Comment:
oh yeah you're trying to convert this to an instance method from the class.  so you gotta go back up there and change `def self.validate_json` to `def validate_json`